### PR TITLE
updated plot_pk function

### PR DESCRIPTION
**Pull request by annahaynie**

### CODE REVIEW

**Review by syasini — CHANGES_REQUESTED** [review]
@annahaynie Thanks, Anna. I think we forgot to return the fig at the end of the plot function.

**Comment by annahaynie** [comment]
@syasini I thought we decided that the "return fig" at the end was redundant and caused the plot to show twice? I may be misremembering

**Comment by syasini** [comment]
@annahaynie No, you're absolutely right. It does behave like that in the jupyter notebook, but I think @timothydmorton wanted the function to return an instance of matplotlib. At least that's what's implemented in the current test. Maybe he can clarify this after reviewing the commits.

File: lya/data.py [hunk 1]
```diff
@@ -29,34 +30,38 @@ def load_lya_data(filename=lya_data_path):
...
-        return fig
 
-    else:
-        ...  # TODO: assign this to Aryan
+Returns
+-------
+fig : matplotlib Figure instance
+"""
 
+if params is None:
```
Comment (timothydmorton) [inline lya/data.py]: I think you're missing a `data = load_lya_data(filename)` line somewhere at the top of this function that was here before.

File: lya/data.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from .theory import get_lcdm_pk
 import numpy as np
 import matplotlib.pyplot as plt
+from theory import get_theory_pk
```
Comment (timothydmorton) [inline lya/data.py]: this should be `.theory` (or have the first line be `from .theory import get_lcdm_pk, get_theory_pk`)

File: lya/data.py [hunk 3]
```diff
@@ -29,34 +30,38 @@ def load_lya_data(filename=lya_data_path):
...
 
-    Returns
-    -------
-    fig : matplotlib Figure instance
-    """
+    return data
 
-    data = load_lya_data(filename)
+def plot_pk(params=None, filename=lya_data_path, color='k',marker='o',ms=5,ls='',**kwargs):
+"""Plot p(k) power spectrum data and theory, with panel for residuals
```
Comment (timothydmorton) [inline lya/data.py]: The indentation of this function is missing.  Indentation matters in python (the whole function should be indented; standard is 4 spaces (your text editor should map the tab key to a four-space indent)

File: lya/data.py [hunk 4]
```diff
@@ -29,34 +30,38 @@ def load_lya_data(filename=lya_data_path):
...
-    else:
-        ...  # TODO: assign this to Aryan
+Returns
+-------
+fig : matplotlib Figure instance
+"""
 
+if params is None:
+       plt.errorbar(data.x, data.y, data.y_unc, color='k',marker='o',ms=5,ls='', **kwargs)
+
```
Comment (timothydmorton) [inline lya/data.py:55]: In this logic branch, we also want to be returning a `fig` instance.  So at the beginning of this, you can start with `fix, ax = plt.subplots(1, 1)` and then do `ax.errorbar`, as it was before.

**Review by timothydmorton — CHANGES_REQUESTED** [review]

**Comment by timothydmorton** [comment]
In the notebook, if you don't want the figure to display twice, you can end the last line of your cell with a `;`-- that will suppress the output (meaning, it won't display twice).

**Review by timothydmorton — APPROVED** [review]

**Pull request merged**